src/Data/Traversable.c

```diff
@@ -0,0 +1,81 @@
+#include <purescript.h>
+
+PURS_FFI_FUNC_1(Data_Traversable_array1, a, {
+  return PURS_ANY_ARRAY_NEW(purs_vec_new_from_array(1, a));
+})
+
+PURS_FFI_FUNC_2(Data_Traversable_array2, a, b, {
+  return PURS_ANY_ARRAY_NEW(purs_vec_new_from_array(2, a, b));
+})
+
+PURS_FFI_FUNC_3(Data_Traversable_array3, a, b, c, {
+  return PURS_ANY_ARRAY_NEW(purs_vec_new_from_array(3, a, b, c));
+})
+
+PURS_FFI_FUNC_2(Data_Traversable_concat2, xs, ys, {
+  return purs_any_concat(xs, ys);
+})
+
+const purs_any_t * traversable_go(
+    const purs_any_t * apply,
+    const purs_any_t * map,
+    const purs_any_t * pure,
+    const purs_any_t * f,
+    const purs_vec_t * array,
+    purs_any_int_t bot,
+    purs_any_int_t top) {
+
+  purs_any_int_t pivot;
+
+  switch(top - bot) {
+    case 0:
+      return purs_any_app(pure, PURS_ANY_ARRAY_NEW(purs_vec_new()));
+    case 1:
+      return purs_any_app(
+               purs_any_app(map, Data_Traversable_array1$),
```

Comment:
this is correct but leaks the fact that pure-c appends a `$` to all identifiers. `PURS_FFI_FUNC` and friends attempt to hide this. Your code is correct, but I am wondering if we can somehow improve on this? like a `#define PURS_FFI_VAR(NAME) NAME ## $`.